sympy/assumptions/handlers/order.py

```diff
@@ -66,32 +66,88 @@ def Add(expr, assumptions):
         if r is not True:
             return r
 
-        nonpos = 0
-        for arg in expr.args:
-            if ask(Q.negative(arg), assumptions) is not True:
-                if ask(Q.positive(arg), assumptions) is False:
-                    nonpos += 1
-                else:
-                    break
-        else:
-            if nonpos < len(expr.args):
-                return True
+
+        args = [arg for arg in expr.args if not ask(Q.zero(arg), assumptions)]
+        if not args:
+            return False
+
+        inf = neg_inf = pos_inf = neg = npos = nneg = False
+
+        for arg in args:
+            if pos_inf and neg_inf:
+                return None
+
+            _inf = ask(Q.infinite(arg), assumptions)
+            if _inf:
+                inf = True
+
+            n = ask(Q.negative(arg), assumptions)
+            if n:
+                if _inf is not False:
+                    neg_inf = True
+                neg = True
+                continue
+
+            np = ask(Q.nonpositive(arg), assumptions)
+            if np:
+                if _inf is True:
+                    neg_inf = True
+                npos = True
+                continue
+
+            nn = ask(Q.nonnegative(arg), assumptions)
+            if nn:
+                if _inf or (_inf is None and nn is False):
```

Comment:
This second condition is unreachable, just like it was [last time](https://github.com/sympy/sympy/pull/19241/files#r426253363). It looks like you wiped out those changes.

Reply:
I think the part of the code must remain to maintain consistency. However, if you feel some changes must be made, could you please commit those changes?

Reply:
Repeating my comment (and suggestion) from above:

> This second condition is unreachable, `nn` cannot be `False` inside `if nn`
> ```suggestion
>                 if _inf:
> ```

Reply:
It is throwing a Travis CI error

Reply:
Can you show what the error us?

Reply:
Condition removed. The initial travis internal errors were the problem.